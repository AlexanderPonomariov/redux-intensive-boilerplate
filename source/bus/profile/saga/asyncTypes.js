// asyncTypes
export const asyncTypes = Object.freeze({
    UPDATE_NAME_ASYNC:     'UPDATE_NAME_ASYNC',
    UPDATE_AVATAR_ASYNC:   'UPDATE_AVATAR_ASYNC',
    UPDATE_PASSWORD_ASYNC: 'UPDATE_PASSWORD_ASYNC',
});
