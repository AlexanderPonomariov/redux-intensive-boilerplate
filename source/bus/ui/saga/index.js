// Core
import { takeEvery } from 'redux-saga/effects';

// Instruments
import { types } from 'bus/ui/types';
import { callInitializeWorker } from './workers/initialize';

export const uiWatchers = Object.freeze({
    * watchInitialize () {
        yield takeEvery(types.INITIALIZE, callInitializeWorker);
    },
});
