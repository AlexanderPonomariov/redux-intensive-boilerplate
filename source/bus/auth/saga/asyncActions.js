// Core
import { asyncTypes } from './asyncTypes';

export const authActionsAsync = Object.freeze({
    initializeAsync: () => ({
        type: asyncTypes.INITIALIZE_ASYNC,
    }),
    loginAsync: (credentials) => ({
        type:    asyncTypes.LOGIN_ASYNC,
        payload: credentials,
    }),
    authenticateAsync: (token) => ({
        type:    asyncTypes.AUTHENTICATE_ASYNC,
        payload: token,
    }),
    signupAsync: (userData) => ({
        type:    asyncTypes.SIGNUP_ASYNC,
        payload: userData,
    }),
    logoutAsync: () => ({
        type: asyncTypes.LOGOUT_ASYNC,
    }),
});
