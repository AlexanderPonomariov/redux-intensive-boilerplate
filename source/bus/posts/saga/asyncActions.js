// Core
import { asyncTypes } from './asyncTypes';

export const postsActionsAsync = Object.freeze({
    fetchPostsAsync: () => ({
        type: asyncTypes.FETCH_POSTS_ASYNC,
    }),
    createPostAsync: (comment) => ({
        type:    asyncTypes.CREATE_POST_ASYNC,
        payload: comment,
    }),
    removePostAsync: (postId) => ({
        type:    asyncTypes.REMOVE_POST_ASYNC,
        payload: postId,
    }),
    likePostAsync: (postID) => ({ // В лайках — писать айди большими буквами
        type:    asyncTypes.LIKE_POST_ASYNC,
        payload: postID,
    }),
    unlikePostAsync: (postID) => ({
        type:    asyncTypes.UNLIKE_POST_ASYNC,
        payload: postID,
    }),
});
