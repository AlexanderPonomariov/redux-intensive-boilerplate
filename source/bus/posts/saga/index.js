// Core
import { takeEvery } from 'redux-saga/effects';

// Instruments
import { types } from 'bus/posts/types';
import { asyncTypes } from './asyncTypes';
import { callFetchPostsWorker } from './workers/fetchPosts';
import { callCreatePostWorker } from './workers/createPost';
import { callRemovePostWorker } from './workers/removePost';
import { callLikePostWorker } from './workers/likePost';
import { callUnlikePostWorker } from './workers/unlikePost';

export const postsWatchers = Object.freeze({
    * watchFetchPosts () {
        yield takeEvery(types.FETCH_POSTS, callFetchPostsWorker);
    },
    * watchCreatePost () {
        yield takeEvery(asyncTypes.CREATE_POST_ASYNC, callCreatePostWorker);
    },
    * watchRemovePost () {
        yield takeEvery(asyncTypes.REMOVE_POST_ASYNC, callRemovePostWorker);
    },
    * watchLikePost () {
        yield takeEvery(asyncTypes.LIKE_POST_ASYNC, callLikePostWorker);
    },
    * watchUnlikePost () {
        yield takeEvery(asyncTypes.UNLIKE_POST_ASYNC, callUnlikePostWorker);
    },
});
