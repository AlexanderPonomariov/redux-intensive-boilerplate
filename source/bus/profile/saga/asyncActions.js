// Types
import { asyncTypes } from './asyncTypes';

export const profileActionsAsync = Object.freeze({
    updateNameAsync: (newName) => ({
        type:    asyncTypes.UPDATE_NAME_ASYNC,
        payload: newName,
    }),
    updateAvatarAsync: (avatar) => ({
        type:    asyncTypes.UPDATE_AVATAR_ASYNC,
        payload: avatar,
    }),
    updatePasswordAsync: (newPassword) => ({
        type:    asyncTypes.UPDATE_PASSWORD_ASYNC,
        payload: newPassword,
    }),
});
