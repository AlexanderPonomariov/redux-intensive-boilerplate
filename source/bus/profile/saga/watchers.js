// Core
import { all, takeEvery } from 'redux-saga/effects';

// Instruments
import { asyncTypes } from './asyncTypes';
import { callUpdateNameWorker } from './workers/updateName';
import { callUpdateAvatarWorker } from './workers/updateAvatar';
import { callChangePasswordWorker } from './workers/changePassword';

function* watchUpdateName () {
    yield takeEvery(asyncTypes.UPDATE_NAME_ASYNC, callUpdateNameWorker);
}
function* watchUpdateAvatar () {
    yield takeEvery(asyncTypes.UPDATE_AVATAR_ASYNC, callUpdateAvatarWorker);
}
function* watchChangePassword () {
    yield takeEvery(asyncTypes.UPDATE_PASSWORD_ASYNC, callChangePasswordWorker);
}

export function* watchProfile () {
    yield all([watchUpdateName(), watchUpdateAvatar(), watchChangePassword()]);
}
