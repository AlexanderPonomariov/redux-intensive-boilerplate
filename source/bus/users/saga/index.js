// Core
import { takeEvery } from 'redux-saga/effects';

// Instruments
import { asyncTypes } from './asyncTypes';
import { callFetchUsersWorker } from './workers/fetchUsers';

export const usersWatchers = Object.freeze({
    * watchFetchUsers () {
        yield takeEvery(asyncTypes.FETCH_USERS_ASYNC, callFetchUsersWorker);
    },
});
