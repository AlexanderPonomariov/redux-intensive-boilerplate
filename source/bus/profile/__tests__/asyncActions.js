// Core
import { profileActionsAsync } from '../saga/asyncActions';

describe('profile actions async :', () => {
    test('updateNameAsync snapshot', () => {
        expect(
            profileActionsAsync.updateNameAsync(__.userProfile),
        ).toMatchSnapshot();
    });

    test('updateAvatar snapshot', () => {
        expect(
            profileActionsAsync.updateAvatarAsync(__.userProfile),
        ).toMatchSnapshot();
    });

    test('updatePasswordAsync snapshot', () => {
        expect(
            profileActionsAsync.updatePasswordAsync(__.userProfile),
        ).toMatchSnapshot();
    });
});
