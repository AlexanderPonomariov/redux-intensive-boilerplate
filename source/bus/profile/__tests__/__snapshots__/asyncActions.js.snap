// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions async : updateAvatar snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`profile actions async : updateNameAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`profile actions async : updatePasswordAsync snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
